op-program/client/interop/consolidate.go

```diff
@@ -89,11 +97,17 @@ func RunConsolidation(
 			Number:    optimisticBlock.NumberU64(),
 			Timestamp: optimisticBlock.Time(),
 		}
-		consolidatedOutputRoot := progress.OutputRoot
 		if err := checkHazards(deps, candidate, chain.ChainID, execMsgs); err != nil {
 			if !isInvalidMessageError(err) {
 				return eth.Bytes32{}, err
 			}
+			invalidChains[chain.ChainID] = optimisticBlock
+		}
+	}
+
+	var consolidatedChains []eth.ChainIDAndOutput
+	for i, chain := range superRoot.Chains {
+		if optimisticBlock, ok := invalidChains[chain.ChainID]; ok {
```

Comment:
How do we handle the case where invalidating a block on one chain makes another chain invalid?

Reply:
I hadn't considered this. I note the issues you've created tracking this. So I'll fix that case as a follow up.